components/__tests__/Button.test.tsx

```diff
@@ -0,0 +1,15 @@
+import { render, screen } from "@testing-library/react";
+
+import Button from "@/components/Button/Button";
+
+describe("Button", () => {
+  it("renders a button", () => {
+    render(<Button />);
+
+    expect(screen.getByRole("button")).toBeInTheDocument();
+  });
+  it("renders button", () => {
```

Comment:
I would advise you not to use snapshot testing for react components. your other test case already covers the accessibility related markup so the snapshot doesn't add any value. this continues as soon as you add functionality to your button. the normal test will interact with the button but the snapshot test falls short in this regard as well.
long story short: I've never needed snapshot tests for react components. I had used them but I hadn't needed them and they were just annoying. this is also backed by the broad opinion of the community.